order-api/src/main/java/com/example/orderapi/global/exception/PaymentFeignClientErrorDecoder.java

```diff
@@ -0,0 +1,20 @@
+package com.example.orderapi.global.exception;
+
+import org.springframework.http.HttpStatus;
+
+import feign.Response;
+import feign.RetryableException;
+import feign.codec.ErrorDecoder;
+
+public class PaymentFeignClientErrorDecoder implements ErrorDecoder {
+    private final ErrorDecoder decoder = new Default();
+
+    @Override
+    public Exception decode(String s, Response response) {
+        if (HttpStatus.valueOf(response.status()).is5xxServerError()) {
+            return new RetryableException(response.status(), response.reason(), response.request().httpMethod(), (Long) null, response.request());
```

Comment:
```suggestion
            return new RetryableException(response.status(), response.reason(), response.request().httpMethod(),  null, response.request());
```

`null`을 Long 으로 캐스팅하는 이유가 있을까요?!

Reply:
계속 cast 컴파일 에러가 발생해서 추가 했습니다.
저도 왜 컴파일 에러가 발생하는지..... 찾아봤지만 알 수가 없었습니다.